cli/spinner.ts

```diff
@@ -0,0 +1,115 @@
+// Copyright 2018-2023 the Deno authors. All rights reserved. MIT license.
+
+const encoder = new TextEncoder();
+
+const LINE_CLEAR = encoder.encode("\r\u001b[K"); // From cli/prompt_secret.ts
+const COLOR_RESET = "\u001b[0m";
+const DEFAULT_SPEED = 75;
+const DEFAULT_SPINNER = ["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"];
+
+/** Options for {@linkcode Spinner}. */
+export interface SpinnerOptions {
+  /**
+   * The sequence of characters to be iterated through for animation.
+   *
+   * @default {["⠋", "⠙", "⠹", "⠸", "⠼", "⠴", "⠦", "⠧", "⠇", "⠏"]}
+   */
+  spinner?: string[];
+  /** The message to display next to the spinner. */
+  message?: string;
+  /** The speed of the spinner.
+   *
+   * @default {75}
+   */
+  speed?: number;
+  /** The color of the spinner. Defaults to the default terminal color. */
+  color?: string;
+  /**
+   * Whether to cleanup the output if the process is terminated.
+   *
+   * @default {true}
+   */
+  cleanup?: boolean;
+}
+
+/**
+ * A spinner that can be used to indicate that something is loading.
+ */
+export class Spinner {
+  #options: SpinnerOptions;
+
+  #intervalId: number | undefined;
+  #active = false;
+
+  /**
+   * Creates a new spinner.
+   * @param options - The options for the spinner.
+   *
+   * @example
+   * ```ts
+   * const spinner = new Spinner({ message: "Loading..." });
```

Comment:
For example blocks to work, you must also include imports. Example [here](https://github.com/denoland/deno_std/blob/main/collections/aggregate_groups.ts).